qutip/quantum_systems/jaynes_cummings.py

```diff
@@ -0,0 +1,164 @@
+
+def _create_sqrt_coefficient(rate):
+    """Helper function to create sqrt coefficient from decay rate"""
+    if isinstance(rate, Coefficient):
+        # Extract coefficient information and create sqrt version
+        def sqrt_func(t, args):
+            return np.sqrt(rate(t, args))
+            
+        return coefficient(sqrt_func, args={})
+    else:
+        return np.sqrt(rate)
+    
+def jaynes_cummings(
+    omega_c: Union[float, Coefficient] = 1.0,
+    omega_a: Union[float, Coefficient] = 1.0,
+    g: Union[float, Coefficient] = 0.1,
+    n_cavity: int = 10,
+    rotating_wave: bool = True,
+    cavity_decay: Union[float, Coefficient] = 0.0,
+    atomic_decay: Union[float, Coefficient] = 0.0,
+    atomic_dephasing: Union[float, Coefficient] = 0.0,
+    thermal_photons: float = 0.0) -> QuantumSystem:
+    """
+    Create Jaynes-Cummings system
+
+    The Jaynes-Cummings model describes a two-level atom interacting with a
+    single cavity mode. The Hamiltonian is:
+
+    H = omega_c * a_dag * a + (omega_a/2) * sigma_z + g * (a_dag * sigma_minus + a * sigma_plus)  [with RWA]
+    H = omega_c * a_dag * a + (omega_a / 2) * sigma_z + g * (a_dag + a) * (sigma_plus + sigma_minus) [without RWA]
+
+    Parameters:
+    -----------
+    omega_c : float or Coefficient, default=1.0
+        Cavity frequency, can be constant or time-dependent
+    omega_a : float or Coefficient, default=1.0
+        Atomic transition frequency, can be constant or time-dependent
+    g : float or Coefficient, default=0.1
+        Atom-cavity coupling strength, can be constant or time-dependent
+    n_cavity : int, default=10
+        Cavity Fock space truncation (number of photon states)
+    rotating_wave : bool, default=True
+        Whether to apply rotating wave approximation
+    cavity_decay : float or Coefficient, default=0.0
+        Cavity decay rate, kappa (photon loss rate), can be constant or time-dependent
+    atomic_decay : float or Coefficient, default=0.0
+        Atomic decay rate, gamma (spontaneous emission rate), can be constant or time-dependent
+    atomic_dephasing : float or Coefficient, default=0.0
+        Atomic pure dephasing rate, gamma_phi, can be constant or time-dependent
+    thermal_photons : float, default=0.0
+        Mean thermal photon number, n_th (for thermal bath)
+
+    Returns:
+    --------
+    QuantumSystem
+        Configured Jaynes-Cummings system instance
+
+    """
+    # Build operators
+    operators = {}
+
+    # Cavity operators (tensor with 2-level atomic system)
+    operators['a'] = tensor(destroy(n_cavity), qeye(2))
+    operators['a_dag'] = operators['a'].dag()
+    operators['n_c'] = operators['a_dag'] * operators['a']  # Photon number
+
+    # Atomic operators (tensor with cavity system)
+    operators['sigma_minus'] = tensor(qeye(n_cavity), destroy(2))
+    operators['sigma_plus'] = operators['sigma_minus'].dag()
+    operators['sigma_z'] = tensor(qeye(n_cavity), sigmaz())
```

Comment:
if you are now just using sigma_- and sigma_+ in the hamiltonian, these pauli operators are slightly inconsistent with the Hamiltonian. As far as I can see only place you use one is in the dephasing, in which case the overall minus sign you are introducing probably doesn't matter since it gets squared away in the lindblad.  but to avoid some user confusion or errors, you could replace them with custom versions, `operators['sigma_z'] = operators['sigma_plus'] * operators['sigma_minus'] - operators['sigma_minus'] *  operators['sigma_plus']' ?